test/bigquery.ts

```diff
@@ -1291,6 +1323,53 @@ describe('BigQuery', () => {
             assert.ifError
           );
         });
+
+        it('should allow for optional parameter types', done => {
+          bq.createJob = (reqOpts: JobOptions) => {
+            // tslint:disable-next-line no-any
+            assert.strictEqual((reqOpts as any).params, undefined);
+            done();
+          };
+
+          bq.createQueryJob(
+            {
+              query: QUERY_STRING,
+              params: NAMED_PARAMS,
+              types: NAMED_TYPES,
+            },
+            assert.ifError
+          );
+        });
+
+        it('should throw for invalid provided types', done => {
```

Comment:
This test should only be in the describe block for getTypeDescriptorFromProvidedType_.